templates/rust/http/src/handler.rs

```diff
@@ -1,13 +1,14 @@
-use actix_web::{http::Method, HttpRequest, HttpResponse};
+use crate::config::HandlerConfig;
+use actix_web::{http::Method, web, HttpRequest, HttpResponse};
 use log::info;
 
 // Implement your function's logic here
-pub async fn index(req: HttpRequest) -> HttpResponse {
+pub async fn index(req: HttpRequest, config: web::Data<HandlerConfig>) -> HttpResponse {
     info!("{:#?}", req);
     if req.method() == Method::GET {
-        HttpResponse::Ok().body("Hello!\n")
+        HttpResponse::Ok().body(format!("Hello {}!\n", config.name))
     } else {
-        HttpResponse::Ok().body("Thanks!\n")
+        HttpResponse::Ok().body(format!("Thanks {}!\n", config.name))
```

Comment:
This kinda makes me want to add a `name` argument, similar to the events example, in which the `config.name` is only used if the `name` isn't passed.

Reply:
I'm all for improving this example and looked into it. However, I didn't find an easy/simple way to add it. We'd need to use extractors for both query and body (depending whether get or post is used) and possibly another struct. Maybe we should split the two handlers? We would also need to add more tests.
I would definitely evolve this, but maybe we should do it in a separate PR?

Reply:
Yeah, I was thinking there was a `FromRequest` impl for `String` but that returns the whole body. I'm not sure introducing a `Query` param (or whatever) is worth it. Probably out of scope for an example. My bad!